libs/core/src/lib/carousel/carousel.component.ts

```diff
@@ -0,0 +1,315 @@
+import {
+    AfterContentInit,
+    ChangeDetectionStrategy,
+    ChangeDetectorRef,
+    Component,
+    ContentChildren,
+    EventEmitter,
+    HostBinding,
+    Input,
+    OnDestroy,
+    OnInit,
+    Optional,
+    Output,
+    QueryList,
+    ViewEncapsulation
+} from '@angular/core';
+import { Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+import { RtlService } from '../utils/services/rtl.service';
+import { CarouselItemComponent } from './carousel-item/carousel-item.component';
+
+const ICON_PAGE_INDICATOR_LIMIT = 8;
+let carouselUniqueId = 0;
+
+class CarouselActiveItem {
+    constructor(public readonly activeItem: CarouselItemComponent, public readonly slideDirection: number) {}
+}
+
+@Component({
+    selector: 'fd-carousel',
+    templateUrl: './carousel.component.html',
+    styleUrls: ['./carousel.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    encapsulation: ViewEncapsulation.None
+})
+export class CarouselComponent implements OnInit, AfterContentInit, OnDestroy {
+    /** @hidden */
+    @ContentChildren(CarouselItemComponent, { descendants: true })
```

Comment:
I think `@ContentChildren` is supposed to come after `Input()` and `Output()`, right?

Reply:
fixed